src/Configuration/ConfigurationFactory.php

```diff
@@ -254,6 +255,23 @@ private function retrievePhpUnit(SchemaConfiguration $schema, string $configDir)
         return $phpUnit;
     }
 
+    private function retrievePhpStan(SchemaConfiguration $schema, string $configDir): PhpStan
+    {
+        $phpStan = clone $schema->getPhpStan();
+
+        $phpStanConfigDir = $phpStan->getConfigDir();
+
+        if ($phpStanConfigDir === null) {
+            $phpStan->setConfigDir($configDir);
+        } elseif (!Path::isAbsolute($phpStanConfigDir)) {
+            $phpStan->setConfigDir(sprintf(
```

Comment:
maybe instead of using a setter here a factory would be sufficient?

```
$phpstan = $phpstan->withConfigDir(...);
```

This way you don't need to clone it either and if the config dir does not change we do not create a new object reference

Reply:
well, I just copied this from the PHPUnit implementation as is.

But renamed, thanks